modules/currency-l0/src/main/scala/org/tessellation/currency/l0/snapshot/programs/Genesis.scala

```diff
@@ -79,18 +78,26 @@ object Genesis {
       binaryHash <- signedBinary.toHashed.map(_.hash)
       _ <- stateChannelSnapshotClient.send(identifier, signedBinary)(globalL0Peer)
 
-      _ <- lastBinaryHashStorage.set(binaryHash)
-
-      signedIncrementalBinary <- stateChannelSnapshotService.createBinary(signedFirstIncrementalSnapshot)
+      signedIncrementalBinary <- stateChannelSnapshotService.createBinary(signedFirstIncrementalSnapshot, binaryHash)
       incrementalBinaryHash <- signedIncrementalBinary.toHashed.map(_.hash)
       _ <- stateChannelSnapshotClient.send(identifier, signedIncrementalBinary)(globalL0Peer)
 
-      _ <- lastBinaryHashStorage.set(incrementalBinaryHash)
-
       _ <- consensusManager.startFacilitatingAfterRollback(
         signedFirstIncrementalSnapshot.ordinal,
-        signedFirstIncrementalSnapshot,
-        CurrencySnapshotContext(identifier, hashedGenesis.info)
+        CurrencyConsensusOutcome(
+          signedFirstIncrementalSnapshot.ordinal,
+          Facilitators(List(nodeId)),
+          RemovedFacilitators.empty,
+          WithdrawnFacilitators.empty,
+          Finished(
+            signedFirstIncrementalSnapshot,
+            incrementalBinaryHash,
```

Comment:
That's where the lastBinaryHash resides right now - hance no need for the storage.